src/ILVerify/tests/ILTests/DelegateTests.il

```diff
@@ -45,6 +45,22 @@
         ret
     }
 
+    // assignment from Func<int> to Func<byte> is invalid
+    .method private hidebysig instance class [System.Runtime]System.Func`1<uint8>
+          AssignIntFuncToByteFunc_Invalid_StackUnexpected(class [System.Runtime]System.Func`2<int32> input) cil managed
+    {
+        ldarg.1
+        ret
+    }
+
+    // assignment from Func<byte> to Func<int> is invalid
+    .method private hidebysig instance class [System.Runtime]System.Func`1<int32>
+          AssignByteFuncToIntFunc_Invalid_StackUnexpected(class [System.Runtime]System.Func`1<uint8> input) cil managed
+    {
+        ldarg.1
+        ret
+    }
```

Comment:
Also cover arguments and enums here?
- `Action<int8>` to `Action<int32>`
- `Action<int8 enum>` to `Action<int8 enum>`